archunit/src/test/java/com/tngtech/archunit/library/cycle_detection/CyclesAssertion.java

```diff
@@ -0,0 +1,54 @@
+package com.tngtech.archunit.library.cycle_detection;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import org.assertj.core.api.AbstractObjectAssert;
+
+import static java.util.stream.Collectors.toSet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings("UnusedReturnValue")
+class CyclesAssertion extends AbstractObjectAssert<CyclesAssertion, Collection<Cycle<?>>> {
+
+    protected CyclesAssertion(Collection<Cycle<?>> cycles) {
+        super(cycles, CyclesAssertion.class);
+    }
+
+    static CyclesAssertion assertThatCycles(Iterable<? extends Cycle<?>> cycles) {
+        return new CyclesAssertion(ImmutableList.copyOf(cycles));
+    }
+
+    CyclesAssertion hasSize(int size) {
+        assertThat(actual).as(descriptionText()).hasSize(size);
+        return this;
+    }
+
+    CyclesAssertion containsOnly(Cycle<?>... cycles) {
+        hasSize(cycles.length);
+
+        Set<Set<List<?>>> thisOriginsAndTargets = actual.stream().map(it -> toOriginsAndTargets(it.getEdges())).collect(toSet());
+        Set<Set<List<?>>> otherOriginsAndTargets = Arrays.stream(cycles).map(it -> toOriginsAndTargets(it.getEdges())).collect(toSet());
+        assertThat(thisOriginsAndTargets).isSubsetOf(otherOriginsAndTargets);
+        assertThat(otherOriginsAndTargets).isSubsetOf(thisOriginsAndTargets);
```

Comment:
Couldn't this be simplified?
```suggestion
        assertThat(thisOriginsAndTargets).isEqualTo(otherOriginsAndTargets);
```

Reply:
Hmm, I think this might be a refactoring leftover, I also think it should work this way :+1: :slightly_smiling_face: At least if I kept anything from my math studies :joy: